planner/core/logical_plan_builder.go

```diff
@@ -610,12 +610,27 @@ func (b *PlanBuilder) buildProjectionFieldNameFromExpressions(field *ast.SelectF
 	case types.KindNull:
 		// See #4053, #3685
 		return model.NewCIStr("NULL"), nil
-	default:
-		// Keep as it is.
-		if innerExpr.Text() != "" {
-			return model.NewCIStr(innerExpr.Text()), nil
-		}
+	case types.KindBinaryLiteral:
+		// Don't rewrite BIT literal or HEX literals
 		return model.NewCIStr(field.Text()), nil
+	case types.KindInt64:
+		// See #9683
+		// TRUE or FALSE can be a int64
+		if mysql.HasIsBooleanFlag(valueExpr.Type.Flag) {
+			i, _ := valueExpr.GetValue().(int64)
```

Comment:
if i := valueExpr.GetValue().(int64); i == 0{
	return xxx
}else{
	return xxx
}

Reply:
We need to check the BooleanFlag, would you mean
if mysql.HasIsBooleanFlag(valueExpr.Type.Flag) {
if i := valueExpr.GetValue().(int64); i == 0{
return xxx
}else{
return xxx
}
}

Reply:
Yep

Reply:
linting
  ✘  https://revive.run/r#indent-error-flow  if block ends with a return statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary)  
  planner/core/logical_plan_builder.go:622:11

Reply:
Ok, my fault, we should not use `else{`